server/Dockerfile

```diff
@@ -12,13 +12,13 @@ RUN cargo install cargo-watch
 # copy over manifests
 COPY ./Cargo.lock ./Cargo.lock
 COPY ./Cargo.toml ./Cargo.toml
+COPY ./src ./src
 
 # this build step will cache dependencies
 RUN cargo build
-RUN rm src/*.rs
+# RUN rm src/*.rs
 
 COPY ./queries ./queries
-COPY ./src ./src
```

Comment:
I think copying src before the first `cargo build` will break the caching, because everytime the src is changed, dependencies will also be recompiled even if it didn't change.
What we could do instead is to `cargo build` a first time the dependencies (as it was before the PR), and just run a second `cargo build` after copying the dependencies.
So the file should look like:

```
COPY ./Cargo.lock ./Cargo.lock
COPY ./Cargo.toml ./Cargo.toml

RUN cargo build

COPY ./queries ./queries
COPY ./src ./src

RUN cargo build
```

Maybe we will need to keep `RUN rm src/*.rs` if the first cargo build adds files in `src`...
I am not 100% sure about this though. @nockty what do you think ?

Reply:
Yes you're right. The following Dockerfile seems to work for me and it keeps the caching. Can anyone else test it? @quentinverlhac @DavidRobertAnsart 

```
FROM rust:1.39-slim

RUN cargo install cargo-watch

# create a new empty shell project
RUN USER=root cargo new --bin server
WORKDIR /server

# copy over manifests
COPY ./Cargo.lock ./Cargo.lock
COPY ./Cargo.toml ./Cargo.toml

# this build step will cache dependencies
RUN cargo build
RUN rm src/*.rs

# copy source tree
COPY ./queries ./queries
COPY ./src ./src

RUN rm ./target/debug/deps/server*
RUN cargo build

EXPOSE 5000

CMD cargo watch -x run
```

Reply:
It works for me to ! @DavidRobertAnsart could you check and update the PR with Corentin's changes ?
Don't copy paste its entire solution because he has forgotten the `apt-get` lines, which caused trouble on my side.
```
RUN apt-get update
RUN apt-get install -y libssl-dev pkg-config
```